lib/generators/smashing_documentation/install_generator.rb

```diff
@@ -6,49 +6,91 @@ class InstallGenerator < Rails::Generators::Base
+          return
         end
+        configure_smashing_docs
+        using_minitest? ? add_minitest_hooks : add_rspec_hooks
+        generate_docs_template
       end
 
       private
+      def configure_smashing_docs
+        create_config_file
+        add_configuration
+      end
 
-      def update_rails_helper
-        destination = "spec/rails_helper.rb"
-        create_file(destination) unless File.exist?(destination)
-        append_file(destination,
-          "SmashingDocs.config do |c|\n"\
-          "  c.template_file = 'smashing_docs/template.md'\n"\
-          "  c.output_file   = 'smashing_docs/api_docs.md'\n"\
-          "  c.auto_push     = false\n"\
-          "  c.run_all       = true\n"\
-          "end"
-          ) unless File.readlines(destination).grep(/SmashingDocs.config/).any?
-      end
-
-      def update_spec_helper
-        destination = "spec/spec_helper.rb"
-        if File.exist?(destination)
+      def add_rspec_hooks
+        helper = "spec/spec_helper.rb"
+        if File.exist?(helper)
           insert_into_file(
-            destination,
+            helper,
             "\n  config.after(:each, type: :controller) do\n"\
             "    SmashingDocs.run!(request, response, true)\n"\
             "  end\n"\
             " # config.after(:suite) { SmashingDocs.finish! }",
             after: "RSpec.configure do |config|"
-          ) unless File.readlines(destination).grep(/SmashingDocs.finish/).any?
+          ) unless test_hooks_already_setup?(helper)
         end
       end
 
+      def add_minitest_hooks
+        helper = "test/test_helper.rb"
+        append_file(
+          helper,
+          "\nclass ActionController::TestCase < ActiveSupport::TestCase\n"\
+          "  def teardown\n"\
+          "    SmashingDocs.run!(request, response, true)\n"\
+          "  end\n"\
+          "end\n"\
+          "MiniTest::Unit.after_tests { SmashingDocs.finish! }\n"
+        ) unless test_hooks_already_setup?(helper)
+      end
+
       def generate_docs_template
         source = "real_template.md"
         destination = "smashing_docs/template.md"
         copy_file(source, destination) unless File.exist?(destination)
       end
+
+      def create_config_file
+        unless File.exist?(@config_file)
+          create_file(@config_file)
+          append_file(@config_file, "class ActiveSupport::TestCase\n end\n") if using_minitest?
+        end
+      end
+
+      def add_configuration
+        unless config_already_setup?
+          config = "SmashingDocs.config do |c|\n"\
+                   "  c.template_file = 'smashing_docs/template.md'\n"\
+                   "  c.output_file   = 'smashing_docs/api_docs.md'\n"\
+                   "  c.auto_push     = false\n"\
+                   "end\n"
+          if using_minitest?
+            insert_into_file(@config_file, config, after: "class ActiveSupport::TestCase\n")
+          else
+            append_file(@config_file, config)
+          end
```

Comment:
Any time I find myself doing nested conditionals, I always ask myself if there is a way to refactor. It looks like there is an easy way.

```
      def add_configuration
        unless config_already_setup?
          setup_config
        end
      end

      ...

      def setup_config
        config = "SmashingDocs.config do |c|\n"\
                 "  c.template_file = 'smashing_docs/template.md'\n"\
                 "  c.output_file   = 'smashing_docs/api_docs.md'\n"\
                 "  c.auto_push     = false\n"\
                 "end\n"
        if using_minitest?
          insert_into_file(@config_file, config, after: "class ActiveSupport::TestCase\n")
        else
          append_file(@config_file, config)
        end
      end
```